bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+        vertical="middle",
+    )
+    main_table.add_column(
+        f"Trend Data of {id} versions",
+        justify="center",
+    )
+
+    def add_sub_table_row(
+        sub_table: Table,
+        version: str,
+        unsuccessful: Unsuccessful,
+    ):
+        sub_table.add_row(
+            version,
+            str(len(unsuccessful.skipped)),
+            str(len(unsuccessful.errored)),
+            str(len(unsuccessful.failed)),
+        )
+
+    for dialect, report in dialects_trend.items():
+        test = "tests" if report.total_tests != 1 else "test"
+        sub_table = Table(
+            "Version",
+            "Skips",
+            "Errors",
+            "Failures",
+            caption=f"{report.total_tests} {test} ran",
+            show_edge=False,
+        )
+
+        if versions:
+            for version, unsuccessful in report.latest_to_oldest():
+                add_sub_table_row(sub_table, version, unsuccessful)
+        else:
+            implementation = report.implementations[id]
+            version = implementation.version or "latest"
+            unsuccessful = report.unsuccessful(id)
+            add_sub_table_row(sub_table, version, unsuccessful)
+
+        main_table.add_row(dialect.pretty_name, sub_table)
+
+    return main_table
+
+
+def _trend_table_in_markdown_for(
+    id: ConnectableId,
+    versions: Set[str],
+    dialects_trend: dict[Dialect, _report.Report],
+) -> str:
+    rows_data: list[list[str]] = []
+
+    inner_table_columns = [
+        "Version",
+        "Skips",
+        "Errors",
+        "Failures",
+    ]
+
+    def create_inner_table_row(
+        version: str,
+        unsuccessful: Unsuccessful,
+    ):
+        return [
+            version,
+            str(len(unsuccessful.skipped)),
+            str(len(unsuccessful.errored)),
+            str(len(unsuccessful.failed)),
+        ]
+
+    for dialect, report in dialects_trend.items():
+        test = "tests" if report.total_tests != 1 else "test"
+        inner_table_rows: list[list[str]] = []
+        if versions:
+            for version, unsuccessful in report.latest_to_oldest():
+                inner_table_rows.append(
+                    create_inner_table_row(version, unsuccessful),
+                )
+        else:
+            implementation = report.implementations[id]
+            version = implementation.version or "latest"
+            unsuccessful = report.unsuccessful(id)
+            inner_table_rows.append(
+                create_inner_table_row(version, unsuccessful),
+            )
+
+        inner_markdown_table = convert_table_to_markdown(
+            inner_table_columns,
+            inner_table_rows,
+        )
+        row_data = [
+            dialect.pretty_name,
+            inner_markdown_table,
+            f"**{report.total_tests} {test} ran**",
+        ]
+        rows_data.append(row_data)
+
+    final_content: Iterable[str] = []
+    for _, row_data in enumerate(rows_data):
```

Comment:
You are calling enumerate and not using the index? So just don't call enumerate. And this too should just be a list comprehension, don't do `for foo in bar: baz.append(...)` ever basically.